### DISCUSSION TITLE
How to import .def file for Windows executable project?

### DISCUSSION THREAD

**Discussion by RadWolfie** [NationalSecurityAgency/ghidra#5673] [Q&A]
I tried to look in known places in Ghidra to import .def file extension for Windows executable project. Yet, I came up with nothing other than "load pdb file" which isn't what I'm looking for. I have executable file has over 200 export names that need to be named from .def file in order to reverse engineer easier.

 I do see [DefLoader.java](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/util/opinion/DefLoader.java) file do exist yet seems doesn't connect with Windows' executable interface/context menus or something? Or Ghidra doesn't have def file extension import feature support built-in yet?

Another thing I do see in "Supported Formats" info window is "Module Definition (DEF)" except only show as raw binary from import file option which I'm pretty sure is wrong way to do it.

**Comment by ryanmkurtz** [comment]
I believe the `DefLoader` was intended to only work with the `Add To Program` feature, however, it is currently being implemented as a standalone loader.

**Reply by ryanmkurtz** [reply]
I have created #5676 to address this.  

@RadWolfie, if possible, can you attach a PE and corresponding .def file to that ticket so I can fix it quicker?

**Reply by RadWolfie** [reply]
Done, I recently discovered that \<project name\>.ord in the symbols folder doesn't seem to work either? I'm currently using Ghidra 10.2.3 at this time. When I test .ord dump to output .exports file with CreateExportFileForDLL.java script, it does replace default ordinal to symbol name. Then drag-n-drop binary file in Ghidra doesn't consume updated ordinal symbol name. Yet the .exports file do get deleted?
